.github/workflows/check-release.yml

```diff
@@ -0,0 +1,117 @@
+        run: |
+          echo "new_release=${{ env.NEW_RELEASE }}" >> $GITHUB_OUTPUT
+          echo "package_name=${{ env.PACKAGE_NAME }}" >> $GITHUB_OUTPUT
+          echo "package_version=${{ env.PACKAGE_VERSION }}" >> $GITHUB_OUTPUT
+    outputs:
+      new_release: ${{ steps.version_check.outputs.new_release }}
+      package_name: ${{ steps.version_check.outputs.package_name }}
+      package_version: ${{ steps.version_check.outputs.package_version }}
+
+  build-publish:
+    needs: [check-version]
+    if: |
+      needs.check-version.outputs.new_release == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install hatch
+      - name: Extract release notes from ${{needs.check-version.outputs.package_name}}/CHANGELOG.md
+        id: extract
+        run: |
+          formatted_date=$(date +"%Y-%m-%d")
+          python tools/extract_release_notes.py \
+          "${{needs.check-version.outputs.package_name}}/CHANGELOG.md" \
+          "${{needs.check-version.outputs.package_version}} — $formatted_date"
+      - name: Tag main and create release on github
+        run: |
+          ./tools/release.sh mckinsey vizro ${{ secrets.GITHUB_TOKEN }} \
+          ${{needs.check-version.outputs.package_name}} \
+          ${{needs.check-version.outputs.package_version}} release_body.txt
+      - name: Build package
+        run: |
+          cd "${{ needs.check-version.outputs.package_name }}"
+          hatch build
+      - name: Set PyPI token
+        run: |
+          if [ "${{ needs.check-version.outputs.PACKAGE_NAME }}" == "vizro-core" ]; then
+              echo 'PYPI_TOKEN=${{ secrets.VIZRO_PYPI_TOKEN }}' >> $GITHUB_ENV
+          fi
+      - name: Publish distribution 📦 to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          packages-dir: ${{ needs.check-version.outputs.package_name }}/dist
+          password: ${{ env.PYPI_TOKEN }}
+
+  version-bump:
+    needs: [check-version]
+    if: |
+      needs.check-version.outputs.new_release == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install hatch
+      - name: Bump version to next patch
+        run: |
+          formatted_date=$(date +"%Y-%m-%d/%H-%M-%S")
+          cd "${{ needs.check-version.outputs.package_name }}"
+          git checkout -b "release/version_bump_next_minor/${formatted_date}"
+          hatch version patch,dev
+          hatch run docs:changelog
+          hatch run schema
+          hatch run lint || hatch run lint
```

Comment:
But the lint workflow does not modify the code right?

Reply:
It will modify the code, but what I'm thinking of now is why we even need `linting` here? This workflow will run only after merging to `main`, in this case all this checks should be green already. And also here's the logic if first lint fail then run it again and then push the changes is smth we should be avoid, because this check is present in the different workflow.

I think it is better to forget about my fist comment. What I'm thinking of is that this linting step is unnecessary here or anywhere (I saw it in `hatch.toml`) except lint workflow.

Reply:
The reason why I thought it was necessary is because we are modifying files (`hatch version` and `hatch schema`) and we want the new PR that is raised, to be merable immediately.

I tried this out now, and it seems like that it doesn't need linting, but it did for me before. Let me take it out and see